### PR TITLE
Withdraw repay validation

### PR DESCRIPTION
**Pull request by BartDeCaluwe**
This PR will add:

- repay min amount validation
- repay max amount validation
- repay amount required validation
- withdraw min amount validation
- withdraw max amount validation
- withdraw amount required validation

## Checklist

- [ ] I used the [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.
- [ ] Unit tests have been created if a new feature was added.
- [ ] Translations have been added and extracted if a new feature was added.

### CODE REVIEW

File: src/components/DealInteraction.tsx [hunk 1]
```diff
@@ -46,6 +68,26 @@ export const DealInteraction: FunctionComponent<DealInteractionProps> = ({
...
+								}),
+							},
+							{
+								validator(_, value) {
+									return validateMinAmount(value);
+								},
+							},
+							{
+								validator(_, value) {
+									return validateMaxAmount(value);
```
Comment (SushantChandla) [inline src/components/DealInteraction.tsx]: Can we also have the `max amount` for repay?

Comment (BartDeCaluwe) [reply src/components/DealInteraction.tsx]: there is, or do you mean that we should validate on the yet to be repaid part?
image

Comment (SushantChandla) [reply src/components/DealInteraction.tsx]: yes, I mean validate on yet to be repaid part.

File: src/components/DealInteraction.tsx [hunk 2]
```diff
@@ -16,6 +17,27 @@ export const DealInteraction: FunctionComponent<DealInteractionProps> = ({
 	const intl = useIntl();
 	const [form] = Form.useForm();
 
+	const validateMinAmount = (value): Promise<void> => {
+		const validationMessage = intl.formatMessage({
```
Comment (RuvendeGroote) [inline src/components/DealInteraction.tsx]: extract to MESSAGES

File: src/components/DealInteraction.tsx [hunk 3]
```diff
@@ -16,6 +17,27 @@ export const DealInteraction: FunctionComponent<DealInteractionProps> = ({
...
+			defaultMessage: "'amount' needs to be greater than 0",
+			description: "Deal intereaction: min amount validation message",
+		});
+		return validateMinValue(value, 0, validationMessage);
+	};
+
+	const validateMaxAmount = (value): Promise<void> => {
+		const validationMessage = intl.formatMessage(
```
Comment (RuvendeGroote) [inline src/components/DealInteraction.tsx]: extract

Comment (RuvendeGroote) [reply src/components/DealInteraction.tsx]: same everywhere else of course

**Review by RuvendeGroote — CHANGES_REQUESTED** [review]

**Comment by RuvendeGroote** [comment]
`feat: add min amount validation` is not descriptive enough (same with max). added to what?

**Review by RuvendeGroote — APPROVED** [review]

**Pull request merged**